Adding support for multiple users with different access rights and Text Editor.

nickivnv opened the pull request:
This is a bugfix regarding visualization of broken symlinks. 
In case of broken symbolic link  (missing target) ist thrown exception which leads to misbehavior during displaying of affected folder.

Commit:
Fixed broken symbolic links visualisation. If symbolic link was broken the whole folder was not displayed on the client.

nickivnv committed:
Adding "http-proxy-middleware" library to client-react project.
Now the http requests to file API have the Same-origin as client URL.
Necessary for correct handling of cookie sessions in development environment.

nickivnv committed:
if the OS (e.g. Embedded Linux Platform) where File Manager is running, support only one language - usually LANG=en_US(utf8 encoding),
then the file names which are ANSI encoded with characters like - ü,ä,ö are not displayed (including all items in the current folder).
This fix make them visible, but not processable (rename, delete, etc.).

nickivnv committed:
Merge branch 'master' of https://github.com/nickivnv/filemanager

nickivnv committed:
Merge branch 'OpusCapita:master' into master

nickivnv committed:
Bug fix: Unit test with Node v16.13.1 ended with the following errors:
  66 passing (540ms)
  4 failing

  1) Search for files/dirs Various fileContentSubstring Case-insensitive content search by default:
     AssertionError: expected [] to have the same members as [ 'gpl.pdf', 'hello-world.js' ]
      at api-tests/api-tests.spec.js:527:66
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

  2) Search for files/dirs Various fileContentSubstring Case-insensitive content search:
     AssertionError: expected [] to have the same members as [ 'gpl.pdf', 'hello-world.js' ]
      at api-tests/api-tests.spec.js:551:66
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

  3) Search for files/dirs Various fileContentSubstring Case-insensitive content + filename search:
     AssertionError: expected [] to have the same members as [ 'hello-world.js' ]
      at api-tests/api-tests.spec.js:576:66
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

  4) Search for files/dirs Various fileContentSubstring Case-sensitive content search:
     AssertionError: expected [] to have the same members as [ 'gpl.pdf' ]
      at api-tests/api-tests.spec.js:595:66
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

 Solution was to rewrite "async _transform(file, encoding, done)" function to synchronous one.

nickivnv committed:
Merge branch 'master' of https://github.com/nickivnv/filemanager

nickivnv committed:
Merge branch 'OpusCapita:master' into master

nickivnv committed:
Supporting users with different access rights. Adding 'express-session' support to the nodejs server.
Test cases are added for signin/signout requests.
This are only server changes, client changes will follow.

nickivnv committed:
Add Signin Dialog to the components.

nickivnv committed:
Response with http error 419 if session expired.

nickivnv committed:
Add cookie session to the unit test.

nickivnv committed:
Add MultiUserAccessFileManager client component.

baur commented:
Is it possible to enable access to folders by list:

case1
access = {user: 'login1', access: [{folder: 'root'}], type: 'read'}

case2
access =  {user: 'login2', access: [{folder: 'root\folder1'}, {folder: 'root\folder2\subfolder3'}], type: 'write'}

user can access various folders

baur commented:
It would be nice if add JWT authorization capability

Thank you for PR. can you publish it to npm if the authors will not merge this PR?

nickivnv commented:
> Is it possible to enable access to folders by list:
> 
> case1 access = {user: 'login1', access: [{folder: 'root'}], type: 'read'}
> 
> case2 access = {user: 'login2', access: [{folder: 'root\folder1'}, {folder: 'root\folder2\subfolder3'}], type: 'write'}
> 
> user can access various folders

For sure it can be done. It is open source do it yourself.

baur commented:
> For sure it can be done. It is open source do it yourself.  

@nickivnv
@estambakio-sc

Sure, I'll try, though I'm not at that level. But for this requires to merge this PR

nickivnv commented:
install it with --force
I can resolve those problems, but I am waiting for the merge of my PR.
In general it would be nice to use the latest React version, but I am not
aware if the owners are going to do it.


On Tue, Aug 2, 2022 at 9:40 AM Bauyrzhan Yessetov ***@***.***>
wrote:

> Package init error
>
> npm install
> error ->
>
> npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
>
> > postinstall
> > npm run bootstrap
>
> npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
>
> > bootstrap
> > lerna bootstrap
>
> lerna info version 2.8.0
> lerna info Bootstrapping 5 packages
> lerna info lifecycle preinstall
> lerna info Installing external dependencies
> lerna ERR! execute callback with error
> lerna ERR! Error: Command failed: npm install
> lerna ERR! npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
> lerna ERR! npm ERR! code ERESOLVE
> lerna ERR! npm ERR! ERESOLVE unable to resolve dependency tree
> lerna ERR! npm ERR!
> lerna ERR! npm ERR! While resolving: ***@***.***
> lerna ERR! npm ERR! Found: ***@***.***
> lerna ERR! npm ERR! node_modules/react
> lerna ERR! npm ERR!   react@"16.2.0" from the root project
> lerna ERR! npm ERR!
> lerna ERR! npm ERR! Could not resolve dependency:
> lerna ERR! npm ERR! peer react@"0.13.x || 0.14.x || ^15.0.0-0 || 15.x" from ***@***.***
> lerna ERR! npm ERR! node_modules/enzyme
> lerna ERR! npm ERR!   enzyme@"2.9.1" from the root project
> lerna ERR! npm ERR!   peer enzyme@"1.x || 2.x" from ***@***.***
> lerna ERR! npm ERR!   node_modules/chai-enzyme
> lerna ERR! npm ERR!     chai-enzyme@"0.6.1" from the root project
> lerna ERR! npm ERR!
> lerna ERR! npm ERR! Fix the upstream dependency conflict, or retry
> lerna ERR! npm ERR! this command with --force, or --legacy-peer-deps
> lerna ERR! npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
> lerna ERR! npm ERR!
> lerna ERR! npm ERR! See C:\Users\admin\AppData\Local\npm-cache\eresolve-report.txt for a full report.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpusCapita/filemanager/pull/331#issuecomment-1202131400>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYJE3DZYKG47RE7TTD4NELTVXDGGFANCNFSM5RB4GZQQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Nikolay Ivanov *| Software Engineer
*Novexx Solutions GmbH* | Ohmstrasse 3, 85386 Eching, Germany
Tel.: +49 8165 925- <%2B49%208165%20925-289>204 | Fax: +49 8165
<%2B49%208165%20925%20327>4225
E-Mail:* ***@***.***
***@***.***> | www.novexx.com <http://www.novexx.com/> *
Geschäftsführer: Dietrich Barsch, Manfred Borbe
Sitz der Gesellschaft: Eching | Amtsgericht München: HRB 217347
VAT-ID-Nr.: DE815545342 | Niederlassung Schweden VAT: SE502077662001
Niederlassung Italien VAT: 09085650969 | WEEE-Reg.-Nr. DE 37487545

baur commented:
ok, thank you ...

baur commented:
I got the next error:

`filemanager\packages\client-react>npm run build`

error ->

```
npm WARN     dev webpack-cli@"*" from the root project
npm WARN     1 more (@webpack-cli/configtest)
[#######...........] \ idealTree:yargs: timing idealTree:node_modules/yargs Completed in 26ms                           [#######...........] | idealTree:serve-static: timing idealTree:node_modules/@opuscapita/opuscapita-ui/node_modules/serv[#######...........] - idealTree:serve-static: timing idealTree:node_modules/@opuscapita/opuscapita-ui/node_modules/serv[########..........] - idealTree:is-number: timing idealTree:node_modules/has-values/node_modules/is-number Completed in[########..........] \ idealTree:style-loader: timing idealTree:node_modules/size-limit/node_modules/style-loader Completed in 0ms                                                                               [########..........] | idealTree:webpack-bundle-analyzer: timing idealTree:node_modules/size-limit/node_modules/webpack-bundle-analyzer Completed inpm ERR! code ETARGET
npm ERR! notarget No matching version found for @opuscapita/react-filemanager-connector-google-drive-v2@1.1.13.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
```

nickivnv commented:
sorry this is also a question to the mentinors. version 1.1.13 is not
published on npm, but is referenced in the project. change it on your dev.
host to point to your local copy.

On Tue, Aug 2, 2022 at 10:26 AM Bauyrzhan Yessetov ***@***.***>
wrote:

> I got the next error:
>
> filemanager\packages\client-react>npm run build
>
> error ->
>
> npm WARN     dev webpack-cli@"*" from the root project
> npm WARN     1 more ***@***.***/configtest)
> [#######...........] \ idealTree:yargs: timing idealTree:node_modules/yargs Completed in 26ms                           [#######...........] | idealTree:serve-static: timing ***@***.***/opuscapita-ui/node_modules/serv[#######...........] - idealTree:serve-static: timing ***@***.***/opuscapita-ui/node_modules/serv[########..........] - idealTree:is-number: timing idealTree:node_modules/has-values/node_modules/is-number Completed in[########..........] \ idealTree:style-loader: timing idealTree:node_modules/size-limit/node_modules/style-loader Completed in 0ms                                                                               [########..........] | idealTree:webpack-bundle-analyzer: timing idealTree:node_modules/size-limit/node_modules/webpack-bundle-analyzer Completed inpm ERR! code ETARGET
> npm ERR! notarget No matching version found for @***@***.***
> npm ERR! notarget In most cases you or one of your dependencies are requesting
> npm ERR! notarget a package version that doesn't exist.
>
> npm ERR! A complete log of this run can be found in:
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpusCapita/filemanager/pull/331#issuecomment-1202176960>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYJE3D6GS4OFW2RVRX6SA2LVXDLSHANCNFSM5RB4GZQQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Nikolay Ivanov *| Software Engineer
*Novexx Solutions GmbH* | Ohmstrasse 3, 85386 Eching, Germany
Tel.: +49 8165 925- <%2B49%208165%20925-289>204 | Fax: +49 8165
<%2B49%208165%20925%20327>4225
E-Mail:* ***@***.***
***@***.***> | www.novexx.com <http://www.novexx.com/> *
Geschäftsführer: Dietrich Barsch, Manfred Borbe
Sitz der Gesellschaft: Eching | Amtsgericht München: HRB 217347
VAT-ID-Nr.: DE815545342 | Niederlassung Schweden VAT: SE502077662001
Niederlassung Italien VAT: 09085650969 | WEEE-Reg.-Nr. DE 37487545

baur commented:
looks like this becomes a big drama )))

builds with warns

```
filemanager\packages\client-react>npm run build --force
npm WARN using --force Recommended protections disabled.

> @baur/react-filemanager@1.1.14 build
> rimraf lib && cross-env NODE_ENV=production webpack --config ./config/webpack.config.js --hide-modules

(node:117808) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
(Use `node --trace-deprecation ...` to show where the warning was created)
Build completed in 4.379s

Hash: 73bf78b0a35ad421d180
Version: webpack 2.2.1
Time: 4381ms
   Asset     Size  Chunks                    Chunk Names
index.js  1.09 MB       0  [emitted]  [big]  main

WARNING in EnvironmentPlugin - HOST environment variable is undefined.

You can pass an object with default values to suppress this warning.
See https://webpack.js.org/plugins/environment-plugin for example.

WARNING in EnvironmentPlugin - PORT environment variable is undefined.

You can pass an object with default values to suppress this warning.
See https://webpack.js.org/plugins/environment-plugin for example.

WARNING in EnvironmentPlugin - SERVER_URL environment variable is undefined.

You can pass an object with default values to suppress this warning.
See https://webpack.js.org/plugins/environment-plugin for example.

WARNING in EnvironmentPlugin - CLIENT_ID environment variable is undefined.

You can pass an object with default values to suppress this warning.
See https://webpack.js.org/plugins/environment-plugin for example.

WARNING in EnvironmentPlugin - API_KEY environment variable is undefined.

You can pass an object with default values to suppress this warning.
See https://webpack.js.org/plugins/environment-plugin for example.
```

npm [published](https://www.npmjs.com/package/@baur/react-filemanager) without error

but after install to project some packages lacks.
![image](https://user-images.githubusercontent.com/2898361/182346676-cb1d3209-8820-44d3-9d0a-2b694837036f.png)
also svg-icons not istallded 


React client Project error ->
```
Uncaught TypeError: Cannot read properties of undefined (reading 'SERVER_URL')
    at Object.<anonymous> (index.js:14412:1)
    at __nested_webpack_require_797__ (index.js:30:1)
    at Object.<anonymous> (index.js:14567:1)
    at __nested_webpack_require_797__ (index.js:30:1)
    at Object.<anonymous> (index.js:9669:1)
    at __nested_webpack_require_797__ (index.js:30:1)
    at Object.<anonymous> (index.js:32266:1)
    at __nested_webpack_require_797__ (index.js:30:1)
    at index.js:76:1
    at index.js:11:1
```

Please, publish on NPM, I'd like to test your version

nickivnv committed:
Server extended to handle file editing.

nickivnv committed:
Add editor dialog to the front-end.

nickivnv committed:
Integrate edward(web editor based on Ace) into server part.

nickivnv committed:
delete some wrong commit files

nickivnv committed:
Change library version for using on node 12.22.2

nickivnv committed:
Replace Edward(https://github.com/cloudcmd/edward) editor with React-Ace(https://github.com/securingsincity/react-ace),
because of the footprint and some other problems related to Embedded Linux Platforms.

nickivnv committed:
Close icon added in the top-right corner.

nickivnv committed:
close icon position property changed.

nickivnv committed:
Show correct file size and date after modifying.

nickivnv committed:
Open files with double click in 'view' or 'edit' mode depending on access rights.

nickivnv committed:
Limit loaded file size for Editor to 100KB.

nickivnv committed:
File loaded limit size error message made as text resource.

nickivnv committed:
Adding some more language modes to AceEditor.

baur commented:
I can't build by your source code, Please, can you pack it to the docker container with node connector?

nickivnv committed:
bug fix with editor view mode.

nickivnv commented:
This is my local copy of the filemanager with all changes necessary to get
it running:
https://drive.google.com/file/d/12o-9A8RbdcdzYFAfkMEPOcXFlURHy3fH/view?usp=share_link

On Wed, Dec 28, 2022 at 3:10 AM Bauyrzhan Yessetov ***@***.***>
wrote:

> I can't build by your source code, Please, can you pack it to the docker
> container with node connector?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpusCapita/filemanager/pull/331#issuecomment-1366315144>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYJE3D4J3YC3I7WKA55V3EDWPOOQNANCNFSM5RB4GZQQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Nikolay Ivanov *| Software Engineer
*Novexx Solutions GmbH* | Ohmstrasse 3, 85386 Eching, Germany
Tel.: +49 8165 925- <%2B49%208165%20925-289>204 | Fax: +49 8165
<%2B49%208165%20925%20327>4225
E-Mail:* ***@***.***
***@***.***> | www.novexx.com <http://www.novexx.com/> *
Geschäftsführer: Dietrich Barsch, Manfred Borbe
Sitz der Gesellschaft: Eching | Amtsgericht München: HRB 217347
VAT-ID-Nr.: DE815545342 | Niederlassung Schweden VAT: SE502077662001
Niederlassung Italien VAT: 09085650969 | WEEE-Reg.-Nr. DE 37487545

nickivnv committed:
prevent exception on client, if not imported editor mode is used.

baur commented:
> This is my local copy of the filemanager with all changes necessary to get it running: https://drive.google.com/file/d/12o-9A8RbdcdzYFAfkMEPOcXFlURHy3fH/view?usp=share_link

Thank you!!!

nickivnv committed:
Move location bar to the top area. which is usual for most known File Managers (Linux, Windows, etc.)

JanRodewald committed:
Add: Add a plot button to show .plot.html files with plotly in the fm

JanRodewald committed:
Merge pull request #1 from nickivnv/addPlotlySupport

RM #3610  - Add: Add a plot button to show .plot.html files with plotly in the fm

JanRodewald committed:
Reduce sice of plotly used by the react fm

JanRodewald committed:
Add max resize of the plots in fm

JanRodewald committed:
Add dynamic support for plotly CDN

JanRodewald committed:
Change width of plotly graph

JanRodewald committed:
Solved Problem, that the first plot wasn't shown

JanRodewald committed:
Repair dynamic loading of the plotly script

JanRodewald committed:
Update the dynamic read of the plotly library

JanRodewald committed:
Update the Plotly dynamic load

JanRodewald committed:
Plotly - Add possibility to change between single plot view and multiple plot view

JanRodewald committed:
Increase the max size for file to display, because plot files can get big
